tools/flakes/detect_flakes.py

```diff
@@ -13,92 +13,251 @@
       pageToken=None, **query_job['jobReference']).execute(num_retries=3)
   rows = page.get('rows')
   if rows:
     return {row['f'][0]['v']:
-            (row['f'][1]['v'], row['f'][2]['v'], row['f'][3]['v'])
+            Row(poll_strategy=row['f'][1]['v'],
+                job_name=row['f'][2]['v'],
+                build_id=row['f'][3]['v'],
+                timestamp=row['f'][4]['v'])
             for row in rows}
   else:
     return {}
 
 
-def get_new_flakes():
-  last_week_sans_yesterday = get_flaky_tests(-14, -1)
-  last_24 = get_flaky_tests(0, +1)
-  last_week_sans_yesterday_names = set(last_week_sans_yesterday.keys())
-  last_24_names = set(last_24.keys())
-  logging.debug('|last_week_sans_yesterday| =', len(last_week_sans_yesterday_names))
-  logging.debug('|last_24_names| =', len(last_24_names))
-  new_flakes = last_24_names - last_week_sans_yesterday_names
-  logging.debug('|new_flakes| = ', len(new_flakes))
-  return {k: last_24[k] for k in new_flakes}
+def parse_isodate(date_str):
+  return datetime.datetime.strptime(date_str, "%Y-%m-%d").date()
+
+
+def get_new_flakes(args):
+  """The from_date_str argument marks the beginning of the "calibration", used
+  to establish the set of pre-existing flakes, which extends over
+  "calibration_days".  After the calibration period, "reporting_days" is the
+  length of time during which new flakes will be reported.
+
+from
+date
+  |--------------------|---------------|
+  ^____________________^_______________^
+       calibration         reporting
+         days                days
+  """
+  dates = process_date_args(args)
+  calibration_results = get_flaky_tests(dates['calibration']['begin'],
+                                        dates['calibration']['end'])
+  reporting_results = get_flaky_tests(dates['reporting']['begin'],
+                                      dates['reporting']['end'])
+  logging.debug('Calibration results: %s', pprint.pformat(calibration_results))
+  logging.debug('Reporting results: %s', pprint.pformat(reporting_results))
+
+  calibration_names = set(calibration_results.keys())
+  logging.info('|calibration_results (%s, %s]| = %d',
+               dates['calibration']['begin'].isoformat(),
+               dates['calibration']['end'].isoformat(),
+               len(calibration_names))
+  reporting_names = set(reporting_results.keys())
+  logging.info('|reporting_results (%s, %s]| = %d',
+               dates['reporting']['begin'].isoformat(),
+               dates['reporting']['end'].isoformat(),
+               len(reporting_names))
+
+  new_flakes = reporting_names - calibration_names
+  logging.info('|new_flakes| = %d', len(new_flakes))
+  return {k: reporting_results[k] for k in new_flakes}
+
+
+def build_args_parser():
+  import argparse, datetime
+  parser = argparse.ArgumentParser()
+  today = datetime.date.today()
+  a_week_ago = today - datetime.timedelta(days=7)
+  parser.add_argument('--calibration_days', type=int, default=7,
```

Comment:
is it based on today's date     or   now() - 24hours.  Also, is it "today" in pacific time, UTC or my local time?

Reply:
Timezones are all MTV now. You can see that in the "-08" following the time (the standard way to represent shift from UTC)